docs/tutorial/go.rst

```diff
@@ -0,0 +1,351 @@
+    2024-10-04T08:51:39.974Z [go] 2024/10/04 08:51:39 new hello world request
+
+You can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press :kbd:`Ctrl` + :kbd:`C`.
+
+
+Cleanup
+~~~~~~~
+
+Now we have a fully functional rock for a Go application! This concludes
+the first part of this tutorial, so we'll stop the container and remove the
+respective image for now:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+
+Update Go application
+========================================
+
+As a final step, let's update our application. For example,
+we want to add a new ``/time`` endpoint which returns the current time.
+
+Start by opening the ``main.go`` file in a text editor and update the code to
+look like the following:
+
+.. literalinclude:: code/go/main.go.time
+    :language: go
+
+Since we are creating a new version of the application, **open the**
+``rockcraft.yaml`` file and change the ``version`` (e.g. to ``0.2``).
+
+.. note::
+
+    ``rockcraft pack`` will create a new image with the updated code even if you
+    don't change the version. It is recommended to change the version whenever
+    you make changes to the application in the image.
+
+Pack and run the rock using similar commands as before:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-update]
+    :end-before: [docs:docker-run-update-end]
+    :dedent: 2
+
+.. note::
+
+    Note that the resulting ``.rock`` file will now be named differently, as
+    its new version will be part of the filename.
+
+Finally, use ``curl`` to send a request to the ``/time`` endpoint:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:curl-time]
+    :end-before: [docs:curl-time-end]
+    :dedent: 2
+
+The updated application should respond with the current date and time (e.g.
+``2024-06-21 09:47:56``).
+
+.. note::
+
+    If you are not getting the current date and time from the ``/time``
+    endpoint, check the :ref:`troubleshooting-go` steps below.
+
+
+Cleanup
+~~~~~~~
+
+We can now stop the container and remove the corresponding image:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker-updated]
+    :end-before: [docs:stop-docker-updated-end]
+    :dedent: 2
+
+
+Reset your environment
+======================
+
+We've reached the end of this tutorial.
+
+If you'd like to reset your working environment, you can simply run the
+following:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:cleanup]
+    :end-before: [docs:cleanup-end]
+    :dedent: 2
+
+.. collapse:: If using Multipass...
+
+    If you created an instance using Multipass, you can also clean it up.
```

Comment:
"If you created an instance using Multipass" -- I did create an instance with multipass, it's the only option provided in the setup section. Is there another option? If so, maybe we can enrich the setup part. If not, we can reword this, simply delete the VM is fine?

Reply:
I think it depends, some users don't want to use multipass